src/app/hooks/useCarts.tsx

```diff
@@ -30,7 +30,7 @@ export const CartContextProvider = (props: Props)=> {
             return [...ProductsArray, product]
         });
         localStorage.setItem('cartProduct', JSON.stringify([...cartProducts, product]))
-    },[cartProducts])
+    },[])
```

Comment:
Since you are passing `product` into the useCallback as dependencies, then you need to specify the product parameters here. So it should be `[cartProducts]`